include/oneapi/dpl/pstl/hetero/dpcpp/utils_ranges_sycl.h

```diff
@@ -754,6 +757,32 @@ __select_backend(const execution::fpga_policy<_Factor, _KernelName>&, _Ranges&&.
 }
 #endif
 
+// Check the outer view type type to see if we can vectorize. Any non-contiguous inputs (e.g. reverse
+// views, permutation views, etc.) cannot be vectorized. If C++20 ranges are present, then we can
+// use the std::ranges::contiguous_range concept.
+template <typename _Rng>
+struct __is_vectorizable_range
+{
+    constexpr static bool value =
+#if _ONEDPL_CPP20_RANGES_PRESENT && _ONEDPL_CPP20_CONCEPTS_PRESENT
+        std::ranges::contiguous_range<_Rng>;
+#else
+        false;
```

Comment:
So, that trait is false for the contiguous views from `oneapi::dpl::experimental::ranges`?

Reply:
We could pull in nanorange headers to use their contiguous range, but should we add the 19000 line header include at this level to do this? 

Unless we have a specific use-case of experimental range API users who would benefit from this, I would favor only supporting C++20 ranges here.

Reply:
I have added nanorange contiguous range and iterator checks where appropriate.